### ISSUE TITLE
Test "track outside git repo" fails with 51e28193614730498ce78a3c5bf128e8383aa208

### ISSUE THREAD

**Issue by javabrett** [github/git-lfs#1075]
**Disclaimer** - this is a fresh build environment by someone who has not built git-lfs previously.

In my `cibuild`, the track-outside test, which asserts that you cannot `git lfs track '../*.foo'` from the top-level of the test-repo, where `..` is also not a repo, fails (in other words the command succeeds whereas the test asserts failure):

```
test: track outside git repo ...                                   FAILED
-- stdout --
    Tracking *.foo
    Initialized empty Git repository in <redacted>/git-lfs/tmp/test-track.sh-4149/track-outside/.git/
    Tracking *.file
    Tracking ../*.foo
-- stderr --
    + set -e
    + git lfs track '*.foo'
    + '[' -n '' ']'
    + git init track-outside
    + cd track-outside
    + git lfs track '*.file'
    + git lfs track '../*.foo'
    + exit 1
    + test_status=1
-- git trace --
   17:20:48.996693 git.c:561               trace: exec: 'git-lfs' 'track' '*.foo'
   17:20:48.996749 run-command.c:334       trace: run_command: 'git-lfs' 'track' '*.foo'
   17:20:49.041419 git.c:344               trace: built-in: git 'init' 'track-outside'
   17:20:49.043785 git.c:561               trace: exec: 'git-lfs' 'track' '*.file'
   17:20:49.043821 run-command.c:334       trace: run_command: 'git-lfs' 'track' '*.file'
   17:20:49.052885 git.c:561               trace: exec: 'git-lfs' 'track' '../*.foo'
   17:20:49.052948 run-command.c:334       trace: run_command: 'git-lfs' 'track' '../*.foo'
```

If I revert in current `master` commit 51e28193614730498ce78a3c5bf128e8383aa208, which changes relevant code in `command_track.go`, the test passes again (git-lfs refuses to add track in non-git parent).  The only complicating factor I can think of is that `track-outside/../../..` is my `git-lfs` clone, although the interceding directories are obviously not part of that repo.

The changed-code seems satisfied as long as my command is run from within the repo.  That might be the intention based on the commit comment "pass gitattributes patterns (mostly) unchanged".

Is allowing that track path now allowed, making the test incorrect?  Does the test need modifying or removal?

Is there a central CI build hosted anywhere, and is the test failing there?

**Comment by technoweenie** [comment]
[Tests on Travis CI](https://travis-ci.org/github/git-lfs/branches) are passing. What version of Git are you running?

I think the problem is that both Travis CI and I am running tests with `$GIT_LFS_TEST_DIR` set to some other directory. It's difficult to test things from outside of a git repository, if they're all running inside your git-lfs clone.

I'm not totally sure how to handle this. Testing Git stuff from a Git repository is tricky. But, I definitely don't want you to have to jump through hoops just to test your custom builds.

**Comment by sinbad** [comment]
The main problem I've had with not setting $GIT_LFS_TEST_DIR is that any problem, either in configuration or code, that happens during cibuild can 'infect' the git-lfs repo config and make subsequent runs unstable. I wonder if we should just automatically assign a temp dir to $GIT_LFS_TEST_DIR if it's blank to always ensure that the tests are run in an isolated env?

**Comment by technoweenie** [comment]
Ah yup, that's definitely a problem too. I think a tmp directory would be great.

**Comment by javabrett** [comment]
Looks like the independent, non-nested test dir is being tacked in #1106 , so let's see whether that fixes everything here.

**Comment by sinbad** [comment]
I had 2 separate PRs in the pipe for this, I wanted to keep them separate (symlink fixes were needed to support using temp dir on OS X) for traceability. So I've closed #1106 because it just pulled in both my sets of changes into one which I wanted to avoid, and instead I've merged #1091 (symlinks) and #1107 (the follow up of always using temp test dirs).

**Comment by javabrett** [comment]
This is fixed by #1107 3117af8.
